include/libtorrent/disk_io_thread.hpp

```diff
@@ -279,7 +279,7 @@ namespace aux { struct block_cache_reference; }
 		, disk_interface
 		, buffer_allocator_interface
 	{
-		disk_io_thread(io_service& ios, counters& cnt);
+		disk_io_thread(io_service& ios, aux::session_settings const&, counters&);
```

Comment:
I think everything in this file is a great win. No duplication of the settings object